src/main/java/view/LoginPage.java

```diff
@@ -149,10 +141,17 @@ public void keyPressed(KeyEvent e) {
                 JPanel adminPagePanel = adminPage.generatePage(cardLayout, mainPanel);
                 mainPanel.add(adminPagePanel, "adminPage");
                 cardLayout.show(mainPanel, "adminPage");
-                return;
-            }
-            // Switch to the new page when the login button is clicked
-            if(tryLogIn(emailField, passwordField)) {
+            }else if(emailField.getText().contains("@eKomisariat.pl")) {
+                if(tryLogIn(emailField, passwordField)) {
+                    System.out.println("dziala");
+                    HomePage homePage = new PolicemanPage();
+                    JPanel homePagePanel = homePage.generatePage(cardLayout, mainPanel);
+                    mainPanel.add(homePagePanel, "homePage");
+                    cardLayout.show(mainPanel, "homePage");
+                } else {
+                    JOptionPane.showMessageDialog(null, "Podano błędne dane logowania!");
+                }
+            } else if(tryLogIn(emailField, passwordField)) {
                 HomePage homePage = new HomePage();
                 JPanel homePagePanel = homePage.generatePage(cardLayout, mainPanel);
                 mainPanel.add(homePagePanel, "homePage");
```

Comment:
We can extract `homePage` and `homePagePanel` outside of the "if" statements, in a way that the ifs will only decide if it should be a `new HomePage` or a `new PolicemanPage`